hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/bucket/FileIOEngine.java

```diff
@@ -59,6 +65,30 @@ public FileIOEngine(long capacity, String... filePaths) throws IOException {
     this.fileChannels = new FileChannel[filePaths.length];
     this.rafs = new RandomAccessFile[filePaths.length];
     this.channelLocks = new ReentrantLock[filePaths.length];
+    init();
+  }
+
+  @Override
+  public boolean verifyFileIntegrity(byte[] persistentChecksum, String algorithm)
+    throws IOException {
+    try {
+      byte[] calculateChecksum = calculateChecksum(algorithm);
+      if (!Bytes.equals(persistentChecksum, calculateChecksum)) {
+        throw new IOException("The persistent checksum is " + Bytes.toString(persistentChecksum) +
+          ", but the calculate checksum is " + Bytes.toString(calculateChecksum));
+      }
+      return true;
+    } catch (IOException ioex) {
+      LOG.error("File verification failed because of ", ioex);
+      reinit();
```

Comment:
Why this reinit only in this case?   In case of checksum mismatch no such thing happening no?